pkg/util/urlutil/origin.go

```diff
@@ -0,0 +1,13 @@
+package urlutil
+
+import (
+	"net/url"
+)
+
+func ExtractOrigin(u *url.URL) *url.URL {
+	newU := *u
+	newU.Path = ""
+	newU.RawQuery = ""
+	newU.Fragment = ""
+	return &newU
+}
```

Comment:
According to https://pkg.go.dev/net/url#URL and https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin, we only need to copy scheme, opaque and host.

```golang
func ExtractOrigin(u *url.URL) *url.URL {
  return &url.URL{
    Scheme: u.Scheme,
    Opaque: u.Opaque,
    Host: u.Host,
  }
}
```

Write a test for this function.